app/models/user_badge.rb

```diff
@@ -19,15 +29,58 @@ class UserBadge < ActiveRecord::Base
   after_create do
     Badge.increment_counter 'grant_count', self.badge_id
     UserStat.update_distinct_badge_count self.user_id
+    UserBadge.update_featured_ranks! self.user_id
     DiscourseEvent.trigger(:user_badge_granted, self.badge_id, self.user_id)
   end
 
   after_destroy do
     Badge.decrement_counter 'grant_count', self.badge_id
     UserStat.update_distinct_badge_count self.user_id
+    UserBadge.update_featured_ranks! self.user_id
     DiscourseEvent.trigger(:user_badge_removed, self.badge_id, self.user_id)
   end
 
+  def self.ensure_consistency!
+    self.update_featured_ranks!
+  end
+
+  def self.update_featured_ranks!(user_id = nil)
+    query = <<~SQL
+      WITH featured_tl_badge AS -- Find the best trust level badge for each user
+      (
+        SELECT user_id, max(badge_id) as badge_id
+        FROM user_badges
+        WHERE badge_id IN (1,2,3,4)
+        #{"AND user_id = #{user_id}" if user_id}
```

Comment:
```suggestion
        #{"AND user_id = #{user_id.to_i}" if user_id}
```
Protect ourselves against injection in the future?